src/components/sbb-tooltip/sbb-tooltip.tsx

```diff
@@ -0,0 +1,433 @@
+      this._setTooltipFocus();
+      this._attachWindowEvents();
+    } else if (event.animationName === 'hide') {
+      this._isDismissing = false;
+      this._presented = false;
+      this._dialog.firstElementChild.scrollTo(0, 0);
+      this._dialog.close();
+      this.didDismiss.emit();
+      this._windowEventsController?.abort();
+    }
+  }
+
+  // Set focus on the first focusable element.
+  private _setTooltipFocus(): void {
+    this._prevFocusedElement = document.activeElement as HTMLElement;
+    this._firstFocusable = this._element.querySelector(IS_FOCUSABLE_QUERY);
+
+    if (this._openedByKeyboard) {
+      this._firstFocusable.focus();
+    } else {
+      // Focusing sbb-tooltip__content in order to provide a consistent behavior in Safari where else
+      // the focus-visible styles would be incorrectly applied
+      this._tooltipContentElement.tabIndex = 0;
+      this._tooltipContentElement.focus();
+      this._element.addEventListener(
+        'blur',
+        () => this._tooltipContentElement.removeAttribute('tabindex'),
+        {
+          once: true,
+        }
+      );
+    }
+  }
+
+  private _setTooltipPosition(): void {
+    if (!this._dialog || !this._triggerElement) {
+      return;
+    }
+
+    const tooltipPosition = getElementPosition(this._dialog, this._triggerElement, {
+      verticalOffset: VERTICAL_OFFSET,
+      horizontalOffset: HORIZONTAL_OFFSET,
+      centered: true,
+    });
+
+    this._alignment = tooltipPosition.alignment;
+
+    const arrowXPosition =
+      this._triggerElement.getBoundingClientRect().left -
+      tooltipPosition.left +
+      this._triggerElement.clientWidth / 2 -
+      8; // half the size of the tooltip arrow
+
+    this._element.style.setProperty('--sbb-tooltip-position-x', `${tooltipPosition.left}px`);
+    this._element.style.setProperty('--sbb-tooltip-position-y', `${tooltipPosition.top}px`);
+    this._element.style.setProperty('--sbb-tooltip-arrow-position-x', `${arrowXPosition}px`);
+  }
+
+  public render(): JSX.Element {
+    const closeButton = (
+      <span class="sbb-tooltip__dismiss">
+        <sbb-button
+          accessibility-label="Close"
+          variant="secondary"
+          size="m"
+          type="button"
+          iconName="cross-small"
+          onClick={() => this.dismiss()}
+          ref={(closeBtn) => (this._lastFocusable = closeBtn)}
+        ></sbb-button>
+      </span>
+    );
+
+    return (
+      <Host
+        class={{
+          'sbb-tooltip--presented': this._presented,
+          'sbb-tooltip--presenting': this._isPresenting,
+        }}
+      >
+        <dialog
+          onAnimationEnd={(event: AnimationEvent) => this._onTooltipAnimationEnd(event)}
+          ref={(dialogRef) => (this._dialog = dialogRef)}
+          class={{
+            'sbb-tooltip': true,
+            'sbb-tooltip--dismissing': this._isDismissing,
+            [`sbb-tooltip--${this._alignment?.vertical || 'below'}`]: true,
```

Comment:
to make the class name it better searchable maybe we should use the approach to hard code every possiblity? should also be moved to <Host>.